docs/resources/user_role.md

```diff
@@ -0,0 +1,57 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "datadog_user_role Resource - terraform-provider-datadog"
+subcategory: ""
+description: |-
+  Provides a Datadog UserRole resource. This can be used to create and manage Datadog user_role.
+---
+
+# datadog_user_role (Resource)
+
+Provides a Datadog UserRole resource. This can be used to create and manage Datadog user_role.
+
+## Example Usage
+
+```terraform
+resource "datadog_role" "foo" {
+  name = "foo"
+  permission {
+    id = data.datadog_permissions.bar.permissions.monitors_downtime
+  }
+  permission {
+    id = data.datadog_permissions.bar.permissions.monitors_write
+  }
+}
+
+resource "datadog_user" "foo" {
+  email = "new@example.com"
+}
+
+# Create new team_membership resource
+resource "datadog_user_role" "foo" {
+  team_id = datadog_role.foo.id
```

Comment:
```suggestion
  role_id = datadog_role.foo.id
```
typo?

Reply:
hah! good catch lol, forgot to replace it when i copied it over 💀